pkg/oktaapi/client_wrapper.go

```diff
@@ -176,3 +201,42 @@ func (cw *ClientWrapper) SearchCommonNameContains(ctx context.Context, searchTer
 	}
 	return users, nil
 }
+
+// SearchCommonNameContainsExhaustive searches for a user by their First/Last name in Okta.
+// It doesn't validate if a user is currently active, which allows us to search for users no longer at CMS
+func (cw *ClientWrapper) SearchCommonNameContainsExhaustive(ctx context.Context, searchTerm string) ([]*models.UserInfo, error) {
+	logger := appcontext.ZLogger(ctx)
+
+	// Sanitize searchTerm for \ and ". These characters cause Okta to error.
+	filterRegex := regexp.MustCompile(`[\\"]`)
+	searchTerm = filterRegex.ReplaceAllString(searchTerm, "")
+
+	// profile.SourceType can be EUA, EUA-AD, or cmsidm
+	// the first 2 represent EUA users, the latter represents users created directly in IDM
+	// Okta API only supports matching by "starts with" (sw) or strict equality and not wildcards or "ends with"
+	isFromEUA := fmt.Sprintf(`(profile.SourceType eq "%v" or profile.SourceType eq "%v")`, euaSourceType, euaADSourceType)
+	nameSearch := fmt.Sprintf(`(profile.firstName sw "%v" or profile.lastName sw "%v" or profile.displayName sw "%v")`, searchTerm, searchTerm, searchTerm)
+	searchString := fmt.Sprintf(`%v and %v`, isFromEUA, nameSearch)
+	search := query.NewQueryParams(query.WithSearch(searchString))
+
+	searchedUsers, _, err := cw.oktaClient.User.ListUsers(ctx, search)
+	if err != nil && !errors.Is(err, context.Canceled) {
+		logger.Error("Error searching Okta users", zap.Error(err), zap.String("searchTerm", searchTerm))
+		return nil, err
+	}
+
+	users := []*models.UserInfo{}
+	for _, user := range searchedUsers {
+		profile, err := cw.parseOktaProfileResponse(ctx, user.Profile)
+		if err != nil {
+			return nil, err
+		}
+
+		// If we find EUA users that have logins longer than 4 characters, they're a test user (don't add them to the array)
+		if (profile.SourceType == euaSourceType || profile.SourceType == euaADSourceType) && len(profile.Login) > 4 {
```

Comment:
can we store the max EUA length (4) somewhere as a const?

Reply:
good call! Thanks Sam, I implemented it.